File: ai-job-portal-backend/candidate-service/src/main/java/com/prahlad/aijobportal/candidateservice/experience/service/ExperienceService.java
package com.prahlad.aijobportal.candidateservice.experience.service;

import com.prahlad.aijobportal.candidateservice.experience.dto.request.ExperienceRequest;
import com.prahlad.aijobportal.candidateservice.experience.dto.response.ExperienceResponse;

import java.util.List;
import java.util.UUID;

public interface ExperienceService {

    ExperienceResponse create(UUID userId, ExperienceRequest request);

    List<ExperienceResponse> getAll(UUID userId);

    ExperienceResponse update(UUID userId, UUID experienceId, ExperienceRequest request);

    void delete(UUID userId, UUID experienceId);
}

File: ai-job-portal-frontend/src/features/profile/services/experience.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { ExperienceRequest, ExperienceResponse } from "@/features/profile/types";

/** Maps 1:1 to candidate-service ExperienceController (/candidate/experience). */
export const experienceService = {
  getAll: () => apiClient.get<ApiResponse<ExperienceResponse[]>>("/candidate/experience").then((res) => res.data),

  create: (payload: ExperienceRequest) =>
    apiClient.post<ApiResponse<ExperienceResponse>>("/candidate/experience", payload).then((res) => res.data),

  update: (experienceId: string, payload: ExperienceRequest) =>
    apiClient
      .put<ApiResponse<ExperienceResponse>>(`/candidate/experience/${experienceId}`, payload)
      .then((res) => res.data),

  delete: (experienceId: string) =>
    apiClient.delete<ApiResponse<null>>(`/candidate/experience/${experienceId}`).then((res) => res.data),
};
